src/interpolator_neon.rs

```diff
@@ -225,6 +156,61 @@ impl NeonInterpolator<f64> {
     }
 }
 
+/// A SSE accelerated interpolator
+pub struct NeonInterpolator<T> where T: NeonSample {
+    sincs: Vec<Vec<T::Sinc>>,
+    length: usize,
+    nbr_sincs: usize,
+}
+
+impl<T> SincInterpolator<T> for NeonInterpolator<T> where T: Sample {
+    /// Calculate the scalar produt of an input wave and the selected sinc filter
+    fn get_sinc_interpolated(&self, wave: &[T], index: usize, subindex: usize) -> T {
+        assert!((index + self.length) < wave.len());
+        assert!(subindex < self.nbr_sincs);
+        unsafe { T::get_sinc_interpolated_unsafe(wave, index, subindex, &self.sincs, self.length) }
+    }
+
+    fn len(&self) -> usize {
+        self.length
+    }
+
+    fn nbr_sincs(&self) -> usize {
+        self.nbr_sincs
+    }
+}
+
+impl<T> NeonInterpolator<T> where T: Sample {
+    /// Create a new NeonInterpolator
+    ///
+    /// Parameters are:
+    /// - `sinc_len`: Length of sinc functions.
+    /// - `oversampling_factor`: Number of intermediate sincs (oversampling factor).
+    /// - `f_cutoff`: Relative cutoff frequency.
+    /// - `window`: Window function to use.
+    pub fn new(
+        sinc_len: usize,
+        oversampling_factor: usize,
+        f_cutoff: f32,
+        window: WindowFunction,
+    ) -> Result<Self, MissingCpuFeatures> {
+        if !is_aarch64_feature_detected!("neon") {
+            return Err(MissingCpuFeatures(FEATURES));
+        }
+
+        assert!(sinc_len % 8 == 0, "Sinc length must be a multiple of 8.");
+        let sincs = make_sincs(sinc_len, oversampling_factor, f_cutoff, window);
+        let sincs = unsafe { <T as NeonSample>::pack_sincs(sincs) };
+
+        Ok(Self {
+            sinc,
+            length: sinc_len,
+            nbr_sincs: oversampling_factor,
+            phantom: PhantomData,
```

Comment:
remove "phantom". This and the sinc typo prevent it from compiling with neon support.